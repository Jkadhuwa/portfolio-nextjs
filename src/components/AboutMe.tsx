import React from "react";

const AboutMe = () => {
  return (
    <div className="flex flex-col md:flex-row bg-gray-900 md:mx-12 my-24 ">
      <div className="flex md:basis-1/2">here</div>
      <div className="md:basis-1/2 p-8 md:p-10 lg:p-16 space-y-8 ">
        <div className="flex justify-start space-y-8">
          <h3 className="text-gray-600 uppercase font-medium  lg:tracking-wide">
            Who is Musinda Kadhuwa
          </h3>
        </div>
        <div className="flex flex-col md:flex-row space-y-6 md:space-y-0">
          <span className="flex mx-auto h-12 w-12 md:w-20 md:h:14 lg:w-14 lg:h-14 bg-indigo-600 rounded-full md:mx-0 ">
            {/* <FontAwesomeIcon icon={["fas", "user"]} color="white" /> */}
          </span>
          <h3 className="flex font-medium text-gray-200 text-2xl md:text-4xl items-center ml-3">
            A Bit More About Me
          </h3>
        </div>
        <div className="flex flex-col">
          <span className="text-gray-300 font-light  mb-8">
            <p className="mb-4">
              In the past 3 years I have been working as a Software Engineer(Web
              Developer) designing and developing fast and scalable APIs and
              responsive frontends.
            </p>
            <p></p>
          </span>
          <button className="w-24 rounded font-medium text-medium  p-2 text-white bg-indigo-700">
            More ...
          </button>
        </div>
      </div>
    </div>
  );
};

export default AboutMe;
