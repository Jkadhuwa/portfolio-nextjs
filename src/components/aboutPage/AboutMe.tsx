import React from "react";
import { useQuery } from "@apollo/client";
import profileOperations from "@graphQl/profileOps";
import { MyInfo, AboutMeSkeleton } from "@components";
import ReactMarkdown from 'react-markdown';
import remarkGfm from "remark-gfm";

interface GetUserInfo {
  profile: { cv: { url: string }, bio: string };
}

export default function AboutMe() {
  const { loading, error, data } = useQuery<GetUserInfo>(profileOperations.Queries.getBio, {
    variables: { id: "u-001" },
  });

  if (error) {
    console.log(error);
    return <AboutMeSkeleton />;
  }

  if (loading || data === undefined) return <AboutMeSkeleton />;
  const { bio, cv:{url}} =  data.profile;

  return (
    <div className="px-12 py-10 ">
      <div className="px-12 py-10">
        <article className="md:text-justify  prose max-w-none prose-h2:mb-2.5 prose-h2:text-[1.6rem] prose-p:text-2xl prose-p:leading-[1.65] text-center">
          <ReactMarkdown remarkPlugins={[remarkGfm]}>{bio}</ReactMarkdown>
        </article>

        <ul className="location grid grid-cols-1 sm:grid-cols-2 mt-6 gap-y-2">
          <MyInfo field="freelance" value="Available" />
          <MyInfo field="residence" value="Nairob, Kenya" />
        </ul>

        <a
          rel="noreferrer"
          href={url}
          target="_blank"
          className="text-gray-300 uppercase text-xl border-2 border-solid border-gray-300 w-56 h-14 rounded-full font-semibold flex items-center justify-center mx-auto mt-36 hover:text-main-orange hover:border-main-orange transition-all duration-200"
        >
          download cv
        </a>
      </div>
    </div>
  );
}
