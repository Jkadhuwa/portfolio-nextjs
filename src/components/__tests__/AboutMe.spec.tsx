import React from "react";
import "@testing-library/jest-dom";
import { render, screen, cleanup } from "@testing-library/react";
import FontAwesomeIcon from "@mocks/@fortawesome/fontAwesomeIcon";
import AboutMe from "@components/AboutMe";

afterEach(cleanup);

beforeAll(() => FontAwesomeIcon());

jest.mock("../../public/logo.png", () => {
  return {
    default: { src: "/public/logo.png", height: 100, width: 100 },
  };
});

const createWrapper = () => {
  return render(<AboutMe />);
};
describe("Test Intro component", () => {
  describe("Snapshot Tests", () => {
    const wrapper = createWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  describe("AboutMe Unit Tests", () => {
    it("Should render aboutme correctly", () => {
      createWrapper();
    });

    it("should have correct heading", () => {
      createWrapper();
      expect(screen.getByText(/A Bit More About Me/i)).toBeInTheDocument();
    });

    it("should have a single button with name More ...", () => {
      createWrapper();
       expect(screen.getByRole('button')).toHaveTextContent('More ...');
       expect(screen.getAllByRole("button").length).toBe(1);
    });
  });
});
