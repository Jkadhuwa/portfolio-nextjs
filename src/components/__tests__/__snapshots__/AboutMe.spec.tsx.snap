// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col md:flex-row bg-gray-900 md:mx-12 my-24 "
      >
        <div
          class="flex md:basis-1/2"
        >
          here
        </div>
        <div
          class="md:basis-1/2 p-8 md:p-10 lg:p-16 space-y-8 "
        >
          <div
            class="flex justify-start space-y-8"
          >
            <h3
              class="text-gray-600 uppercase font-medium lg:tracking-wide"
            >
              Who is Musinda Kadhuwa
            </h3>
          </div>
          <div
            class="flex flex-col md:flex-row space-y-6 md:space-y-0"
          >
            <span
              class="flex mx-auto h-12 w-12 md:w-20 md:h:14 lg:w-14 lg:h-14 bg-indigo-600 rounded-full md:mx-0 "
            />
            <h3
              class="flex font-medium text-gray-200 text-2xl md:text-4xl items-center ml-3"
            >
              A Bit More About Me
            </h3>
          </div>
          <div
            class="flex flex-col"
          >
            <span
              class="text-gray-300 font-light mb-8"
            >
              <p
                class="mb-4"
              >
                In the past 3 years I have been working as a Software Engineer(Web Developer) designing and developing fast and scalable APIs and responsive frontends.
              </p>
              <p />
            </span>
            <button
              class="w-24 rounded font-medium text-medium p-2 text-white bg-indigo-700"
            >
              More ...
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col md:flex-row bg-gray-900 md:mx-12 my-24 "
    >
      <div
        class="flex md:basis-1/2"
      >
        here
      </div>
      <div
        class="md:basis-1/2 p-8 md:p-10 lg:p-16 space-y-8 "
      >
        <div
          class="flex justify-start space-y-8"
        >
          <h3
            class="text-gray-600 uppercase font-medium lg:tracking-wide"
          >
            Who is Musinda Kadhuwa
          </h3>
        </div>
        <div
          class="flex flex-col md:flex-row space-y-6 md:space-y-0"
        >
          <span
            class="flex mx-auto h-12 w-12 md:w-20 md:h:14 lg:w-14 lg:h-14 bg-indigo-600 rounded-full md:mx-0 "
          />
          <h3
            class="flex font-medium text-gray-200 text-2xl md:text-4xl items-center ml-3"
          >
            A Bit More About Me
          </h3>
        </div>
        <div
          class="flex flex-col"
        >
          <span
            class="text-gray-300 font-light mb-8"
          >
            <p
              class="mb-4"
            >
              In the past 3 years I have been working as a Software Engineer(Web Developer) designing and developing fast and scalable APIs and responsive frontends.
            </p>
            <p />
          </span>
          <button
            class="w-24 rounded font-medium text-medium p-2 text-white bg-indigo-700"
          >
            More ...
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
